gordon/main.py

```diff
@@ -83,20 +86,74 @@ def setup(config_root=''):
     return config
 
 
-def _log_or_exit_on_exceptions(errors, debug):
+def _log_or_exit_on_exceptions(base_msg, exc, debug):
     log_level_func = logging.warn
     if not debug:
         log_level_func = logging.error
 
-    base_msg = 'Plugin "{name}" was not loaded:'
-    for name, exc in errors:
-        msg = base_msg.format(name=name)
-        log_level_func(msg, exc_info=exc)
+    if isinstance(exc, list):
+        for exception in exc:
+            log_level_func(base_msg, exc_info=exception)
+    else:
+        log_level_func(base_msg, exc_info=exc)
 
     if not debug:
         raise SystemExit(1)
 
 
+def _assert_required_plugins(plugins, debug):
+    # enricher not required
+    required_providers = {
```

Comment:
Alternatively, I think this might actually be clearer if each plugin was tested individually, rather than added to a dict which gets looped through. I might think if I wanted another plugin to be required, all I had to do was add it here, but actually I would have to add a specific if-block in the loop below as well.  Since each item ends up having a specific if-block anyways, I don't see the value of abstracting it away to a new var.

Reply:
Ah interesting, didn't even occur to me. Thanks for pointing that out!

Reply:
For the lurkers, @descent3me and I talked about simplifying this part of the code and it doesn't seem to be possible. I'm renaming a couple of variables to hopefully improve readability.